addon/components/o-s-s/currency-input.ts

```diff
@@ -0,0 +1,109 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+interface OSSCurrencyInputArgs {
+  currency: string;
+  value: number;
+  onChange(currency: string, value: number): void;
+}
+
+export default class OSSCurrencyInput extends Component<OSSCurrencyInputArgs> {
+  private _currencies = usedCurrencies;
+  @tracked selectedCurrency: Currency;
+  @tracked currencySelectorShown: boolean = false;
+  @tracked filteredCurrencies: Currency[] = this._currencies;
+  @tracked value: number = this.args.value || 0;
+
+  constructor(owner: unknown, args: OSSCurrencyInputArgs) {
+    super(owner, args);
+
+    if (typeof this.args.onChange !== 'function') {
+      throw new Error('[component][OSS::CurrencyInput] The parameter @onChange of type function is mandatory');
+    }
+    this.selectedCurrency = this._currencies[0];
+    this._loadExistingData();
+  }
+
+  private _loadExistingData(): void {
+    if (this.args.currency) {
+      this.selectedCurrency =
+        this._currencies.find((currency: Currency) => currency.code === this.args.currency) || this._currencies[0];
+    }
+  }
+
+  get selectedCurrencySymbol(): string {
+    return this.selectedCurrency.symbol || '—';
+  }
+
+  @action
+  onlyNumeric(event: KeyboardEvent): void {
+    const authorizedInputs = ['Backspace', 'Delete', 'ArrowLeft', 'ArrowRight', 'Tab', 'Shift', 'Control', '.', ','];
+    if (/^[0-9]$/i.test(event.key) || authorizedInputs.find((key: string) => key === event.key)) {
+      this.args.onChange(this.selectedCurrency.code, this.value);
+    } else {
+      event.preventDefault();
+    }
+  }
+
+  @action
+  onSearch(keyword: any): void {
+    this.filteredCurrencies = this._currencies.filter((currency: Currency) => {
+      return (
+        currency.code.toLowerCase().indexOf(keyword.toLowerCase()) !== -1 || currency.symbol.indexOf(keyword) !== -1
+      );
+    });
+  }
+
+  @action
+  onSelect(value: any): void {
+    this.selectedCurrency = value;
+    this.args.onChange(this.selectedCurrency.code, this.value);
+    this.hideCurrencySelector();
+  }
+
+  @action
+  toggleCurrencySelector(e: any): void {
+    e.stopPropagation();
+    this.currencySelectorShown = !this.currencySelectorShown;
+  }
+
+  @action
+  hideCurrencySelector(): void {
+    this.currencySelectorShown = false;
+    this.filteredCurrencies = this._currencies;
+  }
+}
+
+type Currency = {
+  code: string;
+  symbol: string;
+};
+
+const usedCurrencies: Currency[] = [
```

Comment:
It is pertinent to set this const in utils or somethink like that?

Reply:
No use for now, we're not using this set anywhere else !